addons/waypoint/functions/fnc_moveVehicleWp.sqf

```diff
@@ -0,0 +1,135 @@
+
+        _driver enableAI "PATH";
+        _driver domove _pos2;
+        _driver disableAI "AUTOTARGET";
+        _driver disableAI "AUTOCOMBAT";
+        _driver disableAI "TARGET";
+        _driver disableai "COVER";
+        _driver disableai "FSM";
+        _driver setBehaviour "AWARE";
+
+        _LeaderVehicle setvariable [QGVAR(DriverWaitTime),(time + 1)];
+        _LeaderVehicle setDriveOnPath (_LeaderVehicle getvariable QGVAR(TargetArrayMovePosition));
+
+        [{
+        //condition
+            params ["_vehicle","_group"];
+            if !(waypointScript [_group,currentWaypoint _group] == QPATHTOF(functions\fnc_moveVehicleWP.sqf)) exitwith {true};
+            if !(alive (driver _vehicle)) exitwith {true};
+            if (lifestate (driver _vehicle) == "INCAPACITATED") exitwith {true};
+            if (_vehicle distance2d (_vehicle getvariable (QGVAR(TargetArrayMovePosition)) select 1) < 8 ) exitwith {true};
+            //refresh AI from "stuck" or move to updated position
+
+            if (_vehicle getvariable QGVAR(DriverWaitTime) < time) then {
+                    _vehicle setvariable [QGVAR(DriverWaitTime),(time + 2)];
+                    if (speed _vehicle < 4) then {
+                        (driver _vehicle) disableAI "Path";
+                        (driver _vehicle) enableAI "Path";
+                    };
+                    if !(isFormationLeader (driver _vehicle)) then {
+                        (driver _vehicle) doFollow leader (driver _vehicle);
+                        dostop (driver _vehicle);
+                };
+                (driver _vehicle) domove ( _vehicle getvariable QGVAR(TargetArrayMovePosition) select 1);
+                _vehicle setDriveOnPath ( _vehicle getvariable QGVAR(TargetArrayMovePosition));
+                false;
+            }
+            },{
+        //code when true
+            params ["_vehicle"];
+
+            //fixing driver behaviour where he can ignore all move orders if vehicle don't achive setDriveOnPath destinations before doFollow.
+            private _point1 = _vehicle getrelpos [2,0];
+            private _point2 = _vehicle getrelpos [3,0];
+            _vehicle setDriveOnPath [_point1,_point2];
+
+            [{
+                params ["_vehicle"];
+                (driver _vehicle) doFollow (leader (driver _vehicle));
+                (driver _vehicle) domove (getposATL _vehicle);
+                (driver _vehicle) enableAI "all";
+                (driver _vehicle) setVariable ["lambs_danger_disableAI", false];
+                (group (driver _vehicle)) setVariable ["lambs_danger_disablegroupAI", false];
+                _vehicle setvariable [QGVAR(ActiveVehicleMove),false];
+            },[_vehicle],0.2] call cba_fnc_waitAndExecute;
+
+            },[_LeaderVehicle,_group]
+    ] call CBA_fnc_waitUntilAndExecute;
+};
+
+waitUntil {sleep 0.4; !(_LeaderVehicle getvariable [QGVAR(ActiveVehicleMove),false])};
```

Comment:
Oh god, please no `waitUntil` and `sleep`. Can this be rewritten to use `CBA_fnc_waitUntilAndExecute`?

Reply:
Looking at ZEN's code, you could use `_group lockWP true/false`:
https://community.bistudio.com/wiki/lockWP
https://github.com/zen-mod/ZEN/blob/master/addons/ai/functions/fnc_searchBuilding.sqf#L101
https://github.com/zen-mod/ZEN/blob/master/addons/common/XEH_postInit.sqf#L98

So you'd have something like:
```SQF
_group lockWP true;
[{!(_LeaderVehicle getvariable [QGVAR(ActiveVehicleMove),false])}, {
    params ["_group"];
    _group lockWP false;
}, [_group]] call CBA_fnc_waitUntilAndExecute;

true
```

Reply:
No, it's already using as much unscheduled as it can. Scripted waypoint needs a... SCRIPT so must be scheduled and finishes when attached script is finished.

Reply:
Looking at the ZEN's example of Search Building waypoint script, it doesn't seems like it does need to use waitUntil and sleeps if you're using `lockWP`.
https://github.com/zen-mod/ZEN/blob/master/addons/ai/functions/fnc_waypointSearchBuilding.sqf#L27

Although it can stay as is for me if that's a problem.